reference/docs-conceptual/dev-cross-plat/vscode/using-vscode-for-debugging-compiled-cmdlets.md

```diff
@@ -0,0 +1,173 @@
+---
+ms.date:  10/19/2020
+keywords:  powershell,cmdlet,debug
+title:  Using Visual Studio Code for debugging compiled cmdlets
+description: How to set a build task and launch configuration for a PSModule project in .NET Core
+---
+# Using Visual Studio Code for debugging compiled cmdlets
+
+This guide will show you how to interactively debug C# source code for a compiled PowerShell module,
+using Visual Studio Code and the C# extension.
+
+> These steps will show you how to debug your module in PowerShell Core. They will not work for
+Windows PowerShell.
+
+Some familiarity with the Visual Studio Code debugger is assumed.
+
+- For a general introduction to the Visual Studio Code debugger, see [Debugging in Visual Studio Code][].
+
+- For examples of debugging PowerShell script files and modules, see [Using Visual Studio Code for remote editing and debugging][].
+
+This guide follows on from the [Writing Portable Modules][] guide.
+
+## Creating a build task
+
+It is a good idea to build your project automatically before launching a debugging session. This
+ensures that you are debugging the latest version of your code.
+
+This can be achieved with a build task:
+
+1. In the Command Palette, run the `Configure Default Build Task` command
+
+   ![Run Configure Default Build Task](media/using-vscode-for-debugging-compiled-cmdlets/configure-default-build-task.png)
+
+2. In the `Select a task to configure` dialog, choose `Create tasks.json file from template`
+
+3. In the `Select a Task Template` dialog, choose `.NET Core`
+
+This will create a new `tasks.json` file, if one does not exist yet.
+
+To test your build task:
+
+1. In the Command Palette, run the `Run Build Task` command
+
+2. In the `Select the build task to run` dialog, choose `build`
+
+A successful build will not, by default, show output in the terminal pane. If you see output, review
+it. If the output contains the text `Project file does not exist`, then you should edit the
+`tasks.json` file to include the explicit path to the C# project, which can be expressed as
+`"${workspaceFolder}/myModule"` (where `myModule` is the name of the project folder, if applicable).
+
+This needs to go after the `build` entry in the `args` list, as follows:
+
+```json
+    {
+        "label": "build",
+        "command": "dotnet",
+        "type": "shell",
+        "args": [
+            "build",
+            "${workspaceFolder}/myModule",
+            // Ask dotnet build to generate full paths for file names.
+            "/property:GenerateFullPaths=true",
+            // Do not generate summary otherwise it leads to duplicate errors in Problems panel
+            "/consoleloggerparameters:NoSummary",
+        ],
+        "group": "build",
+        "presentation": {
+            "reveal": "silent"
+        },
+        "problemMatcher": "$msCompile"
+    }
+```
+
```

Comment:
I believe that cmdlets are an entry point for many PS scripters into C# development. Therefore, I think it's appropriate to expand on build tasks (which is slightly tangential).

Happy to prune at reviewer's discretion.